prclz/prclz/parcels.py

```diff
@@ -139,13 +158,123 @@ def reunion(no_bldg: gpd.GeoDataFrame,
     to parcels.
     """
     reunioned = no_bldg.copy()
-    orphan_uID = []
-    for orphan in reunioned['geometry']:
-        bid = find_parent_parcel_id(orphan, has_bldg, bldgs_df)
-        orphan_uID.append(bid)
-    reunioned['uID'] = orphan_uID
+    reunioned['uID'] = [find_parent_parcel_id(orphan, has_bldg, bldgs_df) 
+                        for orphan in reunioned['geometry']
+                        ]
     reunioned = pd.concat([reunioned, has_bldg])
     reunioned = reunioned.dissolve(by='uID')
     reunioned.reset_index(inplace=True)
 
     return reunioned
+
+
+def main(
+    blocks_path: Union[Path, str], 
+    buildings_path: Union[Path, str], 
+    output_dir: Union[Path, str], 
+    overwrite: bool = False,
+    ) -> None:
+    """
+    Given a block geojson file and a buildings geojson file, and an
+    output directory, creates the corresponding parcels file 
+    """
+    blocks_path = Path(blocks_path)
+    buildings_path = Path(buildings_path)
+    output_dir = Path(output_dir)
+    
+    gadm = blocks_path.stem.replace("blocks_","")
+    output_path = output_dir / "parcels_{}.geojson".format(gadm)
+    output_path.parent.mkdir(exist_ok=True, parents=True)
+    if (not output_path.is_file() ) or overwrite:
+
+        blocks_gdf = csv_to_geo(blocks_path).rename(columns={'block_geom': 'geometry'})
+        blocks_gdf.crs = "EPSG:4326"
+        blocks_gdf.geometry.crs = "EPSG:4326"
+        start_crs = blocks_gdf.crs
+        bldgs_gdf = gpd.read_file(buildings_path)
+
+        # Map each bldg to block_id
+        bldgs_gdf = gpd.sjoin(bldgs_gdf, blocks_gdf, how='left', op='intersects')
+
+        # Parcelize each block_id -- this could be parellelized if
+        # performance becomes a bottleneck
+        all_parcels = []
+        for i, block_id in enumerate(blocks_gdf['block_id']):        
+            bldgs = bldgs_gdf[bldgs_gdf['block_id']==block_id]['geometry']
+            block = blocks_gdf[blocks_gdf['block_id']==block_id]['geometry'].iloc[0]
+            
+            parcels_gdf = make_parcels(bldgs, block)
+            parcels_gdf['block_id'] = block_id
+            all_parcels.append(parcels_gdf)
+            # if i == 0:
+            #     parcels_gdf = make_parcels(bldgs, block)
+            # else:
+            #     parcels_gdf = pd.concat([parcels_gdf, make_parcels(bldgs, block)])
+        parcels_gdf = pd.concat(all_parcels)
+        parcels_gdf.drop(columns=['uID'], inplace=True)
+        parcels_gdf.to_file(output_path, driver='GeoJSON')
+        info("Save gadm %s parcels at %s", gadm, output_path)
+    else:
+        info("Gadm %s parcels already exist at %s", gadm, output_path)
+
+
+def check_within(
```

Comment:
Might want to call this _check_within if the intention is that it not be used externally.